gym/envs/registration.py

```diff
@@ -706,7 +718,9 @@ def _versions(self, namespace: str, name: str) -> Set[Tuple[int, str]]:
                 lambda version: (
                     int(version),
                     namespace,
-                ),
+                )
+                if isinstance(version, str)
+                else (0, namespace),
```

Comment:
Should probably return `(None, namespace)` here when there isn't a version. Could you also change the return type to `Set[Tuple[Optional[int], str]]`.